### PR TITLE
email notifications endpoint implemented #GCPActive

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

- Added repository, core and controller logic for retrieving a summary of a set of email notifications. 
- English descriptions for each result type is added in NotificationsSummaryService in Core. 

Implemented the "old" way, let's transition to using result object and change namespaces to match features in a seperate PR to limit cognitive load of the QA process. 

## Related Issue(s)
- #267

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] ~~Relevant automated test added (if you find this hard, leave it and we'll help out)~~ #302
- [x] All tests run green

### CODE REVIEW

File: src/Altinn.Notifications.Core/Services/NotificationSummaryService.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public class NotificationSummaryService : INotificationSummaryService
+    {
+        private readonly INotificationSummaryRepository _summaryRepository;
+        private static Dictionary<EmailNotificationResultType, string> _emailResultDescriptions = new()
+            {
+                { EmailNotificationResultType.New, "The email has been created, but has not been picked up for processing yet." },
+                { EmailNotificationResultType.Sending, "The email is being processed and will be attempted sent shortly." },
+                { EmailNotificationResultType.Succeeded, "The email has been accepted by the third party email service and will be sent shortly." },
+                { EmailNotificationResultType.Delivered, "The email was delivered to the recipient. No errors reported, making it likely it was received by the recipient." },
+                {EmailNotificationResultType.Failed, "The email was not sent due to an unspecified failure." },
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Core/Services/NotificationSummaryService.cs:21]:
```suggestion
                { EmailNotificationResultType.Failed, "The email was not sent due to an unspecified failure." },
```

File: src/Altinn.Notifications.Core/Services/NotificationSummaryService.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+namespace Altinn.Notifications.Core.Services
+{
+    /// <summary>
+    /// Implementation of <see cref="INotificationSummaryService"/>
+    /// </summary>
+    public class NotificationSummaryService : INotificationSummaryService
+    {
+        private readonly INotificationSummaryRepository _summaryRepository;
+        private static Dictionary<EmailNotificationResultType, string> _emailResultDescriptions = new()
```
Comment (tba76) [inline src/Altinn.Notifications.Core/Services/NotificationSummaryService.cs:15]: This field could be readonly

File: src/Altinn.Notifications.Persistence/Migration/v0.10/01-setup-functions.sql [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿CREATE OR REPLACE FUNCTION notifications.getemailsummary(
```
Comment (tba76) [inline src/Altinn.Notifications.Persistence/Migration/v0.10/01-setup-functions.sql:1]: This script must be placed in a new folder to be run.

**Review by tba76 — APPROVED** [review]
🍺

**Pull request merged**